### ISSUE TITLE
[BUG] Support for Eclipse 2024-06 (4.32)

### ISSUE THREAD

**Issue by JanMosigItemis** [projectlombok/lombok#3688]
**Describe the bug**
Like https://github.com/projectlombok/lombok/issues/3620 but for Eclipse 4.32.

```
Lombok annotation handler class lombok.eclipse.handlers.HandleBuilder failed
java.lang.NoSuchMethodError: 'void org.eclipse.jdt.internal.compiler.ast.CaseStatement.<init>(org.eclipse.jdt.internal.compiler.ast.Expression, int, int)'
	at lombok.eclipse.Eclipse.createCaseStatement(Eclipse.java:285)
	at lombok.eclipse.handlers.singulars.EclipseJavaUtilListSingularizer.appendBuildCode(EclipseJavaUtilListSingularizer.java:75)
	at lombok.eclipse.handlers.HandleBuilder.generateBuildMethod(HandleBuilder.java:821)
	at lombok.eclipse.handlers.HandleBuilder.handle(HandleBuilder.java:563)
	at lombok.eclipse.HandlerLibrary$AnnotationHandlerContainer.handle(HandlerLibrary.java:106)
	at lombok.eclipse.HandlerLibrary.handleAnnotation(HandlerLibrary.java:237)
	at lombok.eclipse.TransformEclipseAST$AnnotationVisitor.visitAnnotationOnType(TransformEclipseAST.java:269)
	at lombok.eclipse.EclipseNode.traverse(EclipseNode.java:107)
	at lombok.eclipse.EclipseAST.traverseChildren(EclipseAST.java:231)
	at lombok.eclipse.EclipseNode.traverse(EclipseNode.java:74)
```

**To Reproduce**
Load a lombokified project into the workspace and check the error log.

**Expected behavior**
Full support for Eclipse 2024-06 (4.32)

**Version info:**
 - Lombok version: 1.18.32
 - Platform: Eclipse Version: 2024-06 (4.32.0) Build id: 20240606-1231, javac 17.0.11

Fix would be appreciated 👍

**Comment by Rawi01** [comment]
Please try the [edge version](https://projectlombok.org/download-edge)

**Comment by dpolivaev** [comment]
The mentioned edge version page contains no download link.

**Comment by Rawi01** [comment]
![img](https://github.com/projectlombok/lombok/assets/5850477/367855b8-77c8-4552-ae68-e8f4d7a846b9)

**Comment by dpolivaev** [comment]
Wow

**Comment by massimiliano** [comment]
I've tested the edge version with Eclipse 2024-06 and it works. Thank you.

**Comment by JanMosigItemis** [comment]
Tested it with above config and edge version: 1.18.33 (2024-05-17 21:31:34 UTC). Works. Thanks 🍰

**Comment by ingogriebsch** [comment]
I tested the edge version of Friday, June 14, 2024 with Spring STS (Version 4.23.0.RELEASE, Build Id: 202406070037, Revision: d779a8466c4c261d699f49bcbb9dddbe217c0e89) and it works. :)

**Comment by janrieke** [comment]
The fix mentioned here is only for eclipse 2024-03.
For 2024-06, use the edge release (1.18.33). See #3688

**Comment by marshmello275** [comment]
Im facing the issue even after upgrading using edge version.
![Screenshot from 2024-06-19 12-25-55](https://github.com/projectlombok/lombok/assets/42472034/f06fc156-85f4-4c98-b72c-6c6722d6a8a9)

**Comment by cosmacol** [comment]
Since upgrading from Eclipse 2024-03 with Lombok 1.18.32 (edge) to Eclipse 2024-06 with Lombok 1.18.34 (stable), I get sparse errors in the Eclipse annotations side strip related to usage of Lombok-generated constructs (usually related to usage of a missing .builder() method, which should be actually provided by @Builder).

No actual errors in the errors view of Eclipse, and the code starts and debugs normally - the "errors" are only displayed in the current editor annotations.

The behavior is not consistent, in most of the classes it works normally, on other it never works - it seems that, for each class, either it works or not - maybe related to some aspects of the class breaking the Eclipse editor integration of Lombok?

**Comment by abryantsev** [comment]
I also get sporadical `The method builder() is undefined for the type <Name>` message in the editor for the very few classes with the @Builder annotation. No stack traces or errors anywhere. The code is compilable. Eclipse version: 2024-06 (4.32.0), Lombok v1.18.34.

**Comment by famod** [comment]
Since Eclipse 2024-06 is generally supported by Lombok 1.18.34, I'd suggest closing this issue and report individual bugs as individual and well scoped issues like #3706 and #3709.

What you guys could try though is to compile your projects via Maven + ECJ + Lombok. See the zipfile in #3709 as an example.
You _should_ be able to see some more details with that.

**Comment by cosmacol** [comment]
> Since Eclipse 2024-06 is generally supported by Lombok 1.18.34, I'd suggest closing this issue and report individual bugs as individual and well scoped issues like #3706 and #3709.
> 
> What you guys could try though is to compile your projects via Maven + ECJ + Lombok. See the zipfile in #3709 as an example. You _should_ be able to see some more details with that.

In relation to this problem reported in my previous comment: https://github.com/projectlombok/lombok/issues/3688#issuecomment-2198110862 the issue specifically affects, since the upgrade, Eclipse IDE editor integration. The code compiled by Eclipse runs fine, without any errors reported in the Problems pane. My assumption is that, in this case, running ECJ externally would just behave as expected, do you agree?

**Comment by ahoehma** [comment]
I see many many exceptions in eclipse like this 

Lombok annotation handler class lombok.eclipse.handlers.HandleConstructor$HandleAllArgsConstructor failed
Stacktrace: java.lang.NullPointerException

eclipse.buildId=4.24.0.202407191701
java.version=22.0.1
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.springframework.boot.ide.branding.sts4
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.springframework.boot.ide.branding.sts4

I'm using already lombok's latest release:
-javaagent:D:\Dev\sts-4.20.0.RELEASE\plugins\org.projectlombok.agent_1.18.34/lombok.jar

And there is now edge release or?
No edge build has been released since the last [stable release](https://projectlombok.org/download) of lombok.

**Comment by Rawi01** [comment]
The original problem was fixed in lombok 1.18.34. For the javadoc/constructor issue, see #3706.
